src/main/java/jovami/trees/KDTree.java

```diff
@@ -154,10 +156,80 @@ protected E nearestNeighbor(KDNode<E> node, double x, double y, KDNode<E> closes
     }
 
     @Override
-    public List<E> rangeSearch(double x1, double y1, double x2, double y2) {
```

Comment:
Deixa estar este dos 4 doubles como metodo publico e deixa o
`public List<E> rangeSearch(KDNode<E> node,Point2D.Double coordInicial,Point2D.Double coordFinal, Boolean cmpX)` como privado, pq quem for a chamar este metodo nao sabe como/nao consegue passar o node certo.
Em vez disso crias no public os dois `Point2D.Double` que passas como parametro para o privado, juntamente com a `root` da KDTree